src/main/java/com/epam/grid/engine/cmd/CommandArgUtils.java

```diff
@@ -38,67 +39,145 @@
 @NoArgsConstructor(access = AccessLevel.PRIVATE)
 public class CommandArgUtils {
 
-    private static final String DELIMITERS = " \\" + System.lineSeparator();
+    private static final char CR = '\r';
+    private static final char LF = '\n';
+    private static final char TAB = '\t';
+    private static final char SPACE = ' ';
+    private static final char QUOTE = '"';
+    private static final char BACKSLASH = '\\';
 
     /**
      * This method forms a command structure from a string command.
      *
-     * @param command The command from the template.
+     * @param command The command from the template engine.
      * @return The command's structure that is ready for execution.
      */
     public static String[] splitCommandIntoArgs(final String command) {
-        final String[] arr = checkAndMergeQuotesStrings(StringUtils.tokenizeToStringArray(command, DELIMITERS));
-        return Arrays.stream(arr).map(String::trim).filter(s -> !s.isEmpty()).toArray(String[]::new);
-    }
-
-    /**
-     * This method searches for parts of quoted strings that are split into several strings when parsing
-     * and combines them into one string.
-     *
-     * @param commandParts The command from the template after parsing.
-     * @return The command's structure.
-     */
-    private static String[] checkAndMergeQuotesStrings(final String[] commandParts) {
         final List<String> result = new ArrayList<>();
-        final StringBuilder quotedPart = new StringBuilder();
-
-        for (String commandPart : commandParts) {
-            if (commandPart.startsWith(QUOTE) || commandPart.startsWith(APOSTROPHE)) {
-                appendCommandToBuilder(quotedPart, commandPart);
-            } else if (commandPart.endsWith(QUOTE) || commandPart.endsWith(APOSTROPHE)) {
-                appendCommandToBuilder(quotedPart, commandPart);
-                dumpQuotedToResults(quotedPart, result);
-            } else {
-                addQuotedPartToBuilderOrDump(commandPart, quotedPart, result);
+        final StringBuilder token = new StringBuilder();
+
+        boolean isQuote = false;
+        boolean isToken = false;
+        int backslashes = 0;
+
+        for (final char ch : command.toCharArray()) {
+            if (!isToken) {
```

Comment:
it seems that we can get rid of `isToken` if we will allow to add ` ` to token but trim it before add to result, and filter result at the end